### DISCUSSION TITLE
Is it possible to access the offset for where a parsed node appears in the source string?

### DISCUSSION THREAD

**Discussion by broneill** [JSQLParser/JSqlParser#1947] [Q&A] [answered]
When the parsed SQL has a syntax error, the offset of where the error occurred is presented in the error message. I'm doing additional query processing, to make it actually work, and so additional errors can be generated during this phase. Unfortunately, I cannot seem to access the original offsets, and so the error messages are less helpful. Is this feature available, but I just can't find it?

**Comment by manticore-projects** [comment]
Can you share some code please?
I get the error position reliably when using the SQL Panel, see here: http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20M4UwNiDGAuAEBUsCGsBGAoAZgJwPYFtZoRhog

But I am interested only in the first error token and do not understand yer what exactly you try to achieve.

**Reply by broneill** [reply]
I'm implementing a database, and so parsing the SQL is only the first step. A query like "select * from foo" is perfectly legal as far as the parser is concerned, but if "foo" doesn't exist, I generate a new error. In this case, it's obvious where the error is in the query. For more complex expressions, I'd like to identify the exact part of the query which is causing an error.

**Comment by manticore-projects** [comment]
Ok, understood. However, this exceeds the capability of JSQLParser since it is not aware of the actual schema objects.
You would need to get the AST from JSQLParser and then very the nodes against the Database MetaData.

I am still lost what you mean by "original offset" though.

**Comment by manticore-projects** [comment, marked as answer]
Reading again, I think I understand your requirement better: You want to get the position of a particular token (which you throw an error for).

Most, but not all Productions link the AstNode, which should give your this position.

**Reply by broneill** [reply]
I'll take a look...

**Comment by broneill** [comment]
I see it now, thanks!
